### PR TITLE
feat: slots rest api

### LINKED ISSUE #415: Expose information about hosted requests
Currently, there is no endpoint that exposes information about what hosts store, for how long, how much money they will get for it, the state, etc.

### PR DESCRIPTION
**Pull request by AuHau**
Closes #415

### CODE REVIEW

File: codex/contracts/market.nim [hunk 1]
```diff
@@ -52,7 +53,11 @@ method myRequests*(market: OnChainMarket): Future[seq[RequestId]] {.async.} =
   return await market.contract.myRequests
 
 method mySlots*(market: OnChainMarket): Future[seq[SlotId]] {.async.} =
-  return await market.contract.mySlots()
+  let slots = await market.contract.mySlots()
+  debug "Fetched my slots", numSlots=len(slots)
+  # debug "Fetched my slots", numSlots=len(slots), slots
```
Comment (markspanbroek) [inline codex/contracts/market.nim:58]:
```suggestion
```

File: codex/proving.nim [hunk 2]
```diff
@@ -56,6 +56,8 @@ proc prove(proving: Proving, slot: Slot) {.async.} =
   without onProve =? proving.onProve:
     raiseAssert "onProve callback not set"
   try:
+    debug "Proving slot"
+    # debug "Proving slot", slotId=slot.id
```
Comment (markspanbroek) [inline codex/proving.nim:60]:
```suggestion
```

**Review by markspanbroek — APPROVED** [review]
Nice, thanks @AuHau!

File: codex/rest/api.nim [hunk 3]
```diff
@@ -252,6 +252,26 @@ proc initRestApi*(node: CodexNodeRef, conf: CodexConf): RestRouter =
...
+        return RestApiResponse.error(Http503, "Sales unavailable")
+
+      let market = contracts.sales.context.market
+
+      var slots: seq[Slot] = @[]
+
+      let slotIds = await market.mySlots()
+      for slotId in slotIds:
+        if slot =? (await market.getActiveSlot(slotId)):
+          slots.add slot
```
[the remark this answers is not in the document]
Comment (AuHau) [reply codex/rest/api.nim:270]: Sounds like good suggestion! Thanks 👍

File: codex/rest/json.nim [hunk 4]
```diff
@@ -55,6 +55,11 @@ func `%`*(arr: openArray[byte]): JsonNode =
 func `%`*(id: RequestId | SlotId | Nonce | AvailabilityId): JsonNode =
   % id.toArray
 
+func `%`*(obj: StorageRequest | Slot): JsonNode =
+  result = newJObject()
```
Comment (dryajov) [inline codex/rest/json.nim:59]: I would not use the implicit result here (in fact, this is discouraged for various reasons and should be generally avoided https://status-im.github.io/nim-style-guide/language.result.html). Use an explicit `ret` or other variable and return that.

**Review by dryajov — CHANGES_REQUESTED** [review]

File: codex/rest/json.nim (hunk 4 above)
Comment (AuHau) [reply codex/rest/json.nim:59]: Ok! Thanks for the feedback!

Comment (AuHau) [reply codex/rest/json.nim:59]: Btw. there is quite some `result` usage in the codebase. Is there some Nim linter with which we could enforce rules like these?

**Review by markspanbroek — APPROVED** [review]

**Pull request merged**